xcube/webapi/context.py

```diff
@@ -150,15 +150,14 @@ def get_ml_dataset(self, ds_id: str) -> MultiLevelDataset:
         ml_dataset, _ = self._get_dataset_entry(ds_id)
         return ml_dataset
 
-    def get_dataset(self, ds_id: str) -> xr.Dataset:
+    def get_dataset(self, ds_id: str, var_names: Collection[str] = None) -> xr.Dataset:
```

Comment:
From a method with this header, I would expect to get a data set that consists only of the the requested variables (i.e., a variable subset), not that it merely checks whether the variables are present. It is an improvement to the previous method, but at the very least it should be documented.

Reply:
Good point. I'll rename it to `expected_var_names` .